README.md

```diff
@@ -0,0 +1,121 @@
+# ACME_ENV=production
+# ACME_DOMAIN=<your registry domain name FQDN>
+
+# Optional Docker user info
+# This supports a single user/password pair.
+# More complex use-cases can be built using this as a base image.
+# These generated strings might be a good user/password combo for your registry
+# REGISTRYUSER=<long random string>
+# REGISTRYPASS=<long random string>
+
+# HTTP secret must be same among all instances of the same registry
+# Feel free to use the randomly generated secret or create your own
+HTTPSECRET=<long random string>
+
+# Set the storage driver (required)
+# Either `filesystem` or `s3` are supported
+STORAGEDRIVER=filesystem
+
+# Optional s3 bucket details if using s3 storage
+# S3ACCESSKEY=<your s3 access key>
+# S3SECRETKEY=<your s3 secret key>
+# S3REGION=<your s3 region>
+# S3BUCKET=<your s3 bucket>
+
+# Consul discovery via Triton CNS
+CONSUL=consul.svc.<user uuid>.<data center>.cns.joyent.com
+```
+
+### Automatic SSL via LetsEncrypt (ACME)
+
+This project includes support for automatic SSL certificates via Let's Encrypt as implemented in [autopilotpattern/nginx](https://github.com/autopilotpattern/nginx).
+
+After configuring your domain name to resolve to the Nginx instance(s) that front the registry, they will automatically acquire certificates for the given domain, and renew them over time. If you scale to multiple instances of Nginx, they will elect a leader who will be responsible for renewing the certificates.  Any challenge response tokens as well as acquired certificates will be replicated to all Nginx instances. 
+
+You must ensure the domain resolves to the Nginx container(s) that front the registry so that they can respond to the ACME http challenges. Triton users may [refer to this document](https://docs.joyent.com/public-cloud/network/cns/faq#can-i-use-my-own-domain-name-with-triton-cns) for more information on how to insure your domain resolves to your Triton containers.
+
+In the `_env` file, set the `ACME_ENV` and `ACME_DOMAIN` to your desired values:
+
+```bash
+# Optional Let’s Encrypt details
+# This will automatically generate a well-recognized SSL certificate
+ACME_ENV=production # or `staging`
+ACME_DOMAIN=<your registry domain name FQDN>
+```
+
+### Password protection
+
+This implementation supports a single user/password combo to limit access (see [autopilotpattern/docker-registry#3](https://github.com/autopilotpattern/docker-registry/issues/3) for plans to support more users, or feel free to extend this Docker image for your needs).
+
+The `setup.sh` script will generate long random strings to use as the username and password pair, simply uncomment those lines to use them, or change them to the username and password of your choice.
+
+```bash
+# Optional Docker user info
+# This supports a single user/password pair.
+# More complex use-cases can be built using this as a base image.
+# These generated strings might be a good user/password combo for your registry
```

Comment:
Could the comments clarify this a bit for end-user uses? Something more like "doing X and Y is how you would have your Docker client login"?